one_fm/hiring/doctype/duty_commencement/duty_commencement.py

```diff
@@ -17,20 +17,26 @@ def validate(self):
 		if getdate(self.date_of_joining) < getdate():
 			frappe.throw(_("Date of joining cannot be before today"))
 
-		if self.workflow_state == "Applicant Signed and Uploaded" and self.employee:
-			employee_doc = frappe.get_doc("Employee", self.employee)
-			employee_doc.project = self.project
-			employee_doc.site = self.operations_site
-			employee_doc.shift = self.operations_shift
-			employee_doc.date_of_joining = self.date_of_joining
-			employee_doc.save(ignore_permissions=True)
-			self.auto_checkin_candidate()
-		elif self.workflow_state == "Applicant Signed and Uploaded" and not self.employee:
-			frappe.throw(_("No Employee created for {employee_name}. Please create employee from Onboard Employee.".format(employee_name=self.employee_name)))
-
 		self.set_progress()
 		self.update_salary_details_from_job_offer()
 
+	def validate_workflow(self):
+		if self.workflow_state == "Applicant Signed and Uploaded":
+			if self.employee:
+				self.auto_checkin_candidate()
```

Comment:
@pjamsheer Let's fetch employee from Onboard Employee itself. We can remove the employee field in Duty Commencement. Also, we can remove code that sets this employee field in Onboard Employee while creating Employee.

Reply:
@Ehtesham599  So instead of `if self.employee` we need to check `if frappe.db.get_value('Onboard Employee', self.onboard_employee, 'employee')`

Reply:
Yes. We can simply do that. If no employee is set, simply throw error saying "No employee created".

Reply:
No, We dont want to throw, just need to notify that you have to create Employee from Onboard Employee, since we can create Employee only if the Duty Commencement Signed and Uploaded.

Reply:
But we cannot submit Duty Commencement without creating the employee. Just need to make sure that the employee is created before submitting.

Reply:
Okay, Let me add those changes also

Reply:
Yeah sure. Let me know once its done.

Reply:
@Ehtesham599 Done